src/background.js

```diff
@@ -360,7 +405,30 @@ const rollout = {
 const setup = {
   async start() {
     const isAddonDisabled = await rollout.getSetting("doh-rollout.disable-heuristics", false);
-    const runAddon = await browser.experiments.preferences.getBoolPref("doh-rollout.enabled", false);
+    const runAddonPref = await browser.experiments.preferences.getBoolPref("doh-rollout.enabled", false);
+    const runAddonBypassPref = await browser.experiments.preferences.getBoolPref("doh-rollout.self-enabled", false);
+    const runAddonLocalStorage = await rollout.getSetting("doh-rollout.doorhanger-decision", false);
+    const remoteDisableAddon = await browser.experiments.preferences.getBoolPref("doh-rollout.remote-disable", false);
```

Comment:
@mythmon Note that this pref is not being listened for. (L450 is constantly listening for `doh-rollout.enabled`). Would this be an issue if we deployed the pref via Normandy?

Reply:
This would be a problem. Storing the value of a preference and not reacting to it changing in the pref system is a big no-no.

In the best case, this would mean that we would only respond to setting the remote disable switch on a reboot. In the worst case, this would be a race condition and make this preference flaky.

In general, you should either

1. Never store the value of a preference, and pull it from the preference system any time you need it.
2. Or store the value and update the stored value via a preference listener.

In this case I think we want to have a listener that will disable the add-on immediately if this preference turns to true.

Reply:
Added the requested listener and logic to flip the self-enabled pref. This means as long as Normandy doesn't push the `doh-rollout.enabled` pref, the add-on will remain off.